lib/iris/_lazy_data.py

```diff
@@ -93,6 +93,8 @@ def as_lazy_data(data, chunks=_MAX_CHUNK_SIZE):
         The input array converted to a dask array.
 
     """
+    if isinstance(data, ma.core.MaskedConstant):
+        data = ma.masked_array([data], mask=data.mask)
```

Comment:
@lbdreyer Perhaps my point was lost in the example, but shouldn't line 97 be
```python
data = ma.masked_array(data, mask=data.mask)
```
instead of 
```python
data = ma.masked_array([data], mask=data.mask)
```